plugins/plugin-hosted/codenvy-hosted-ext-factory/src/main/java/com/codenvy/ide/factory/client/utils/FactoryProjectImporter.java

```diff
@@ -132,6 +132,10 @@ public void apply(List<ProjectConfigDto> projectConfigs) throws OperationExcepti
                 Set<String> projectNames = new HashSet<>();
                 String createPolicy = factory.getPolicies() != null ? factory.getPolicies().getCreate() : null;
                 for (ProjectConfigDto projectConfig : projectConfigs) {
+                    if (projectConfig.getSource().getLocation() == null) {
```

Comment:
As I understand For It can't because source is mandatory parameter in factory https://github.com/eclipse/che/blob/CODENVY-365/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/shared/dto/ProjectConfigDto.java#L80-80

But mb it is better to make source optional when project is subproject. Instead of making optional type and location. WDYT?